internal/core/domain/wallet_repository.go

```diff
@@ -63,23 +63,23 @@ type WalletRepository interface {
 	// its basic info.
 	// Generates a WalletAccountCreated event if successfull.
 	CreateAccount(
-		ctx context.Context, accountName string, birthdayBlock uint32,
+		ctx context.Context, rootPathPurpose, label string, birthdayBlock uint32,
```

Comment:
Not necessary to provide `rootPathPurpose` as arg since it can be extracted from wallet's RootPath.
```suggestion
		ctx context.Context, label string, birthdayBlock uint32,
```